File: api/global-session.py
import json
import os
from supabase import create_client, Client

# Inicializa o cliente Supabase
url = os.environ.get("SUPABASE_URL")
key = os.environ.get("SUPABASE_ANON_KEY")
supabase: Client = create_client(url, key)

def handler(request):
    if request.method == "POST":
        try:
            data = json.loads(request.body)
        except:
            data = {}
        
        global_session_id = data.get('global_session_id', 'global_default')
        
        # Verificar se sessão existe
        session_data = supabase.table("user_sessions").select("*").eq("session_id", global_session_id).execute()
        
        if not session_data.data:
            # Criar sessão
            supabase.table("user_sessions").insert({
                "session_id": global_session_id
            }).execute()
            
            # Criar playlist padrão
            supabase.table("playlists").insert({
                "name": "default",
                "user_id": global_session_id,
                "items": []
            }).execute()
            
            # Buscar sessão criada
            session_data = supabase.table("user_sessions").select("*").eq("session_id", global_session_id).execute()
        
        return {
            "statusCode": 200,
            "headers": {
                "Content-Type": "application/json",
                "Access-Control-Allow-Origin": "*",
                "Access-Control-Allow-Methods": "GET, POST, PUT, DELETE, OPTIONS",
                "Access-Control-Allow-Headers": "Content-Type"
            },
            "body": json.dumps({
                "success": True,
                "session": session_data.data[0]
            })
        }
    
    elif request.method == "OPTIONS":
        return {
            "statusCode": 200,
            "headers": {
                "Access-Control-Allow-Origin": "*",
                "Access-Control-Allow-Methods": "GET, POST, PUT, DELETE, OPTIONS",
                "Access-Control-Allow-Headers": "Content-Type"
            },
            "body": ""
        }
    
    return {
        "statusCode": 405,
        "body": json.dumps({"error": "Method not allowed"})
    }

File: api/global-session.js
import { createClient } from '@supabase/supabase-js';

const supabaseUrl = process.env.SUPABASE_URL;
const supabaseKey = process.env.SUPABASE_ANON_KEY;
const supabase = createClient(supabaseUrl, supabaseKey);

export default async function handler(req, res) {
  if (req.method === 'POST') {
    let data = {};
    try {
      data = req.body;
    } catch {
      data = {};
    }

    const globalSessionId = data.global_session_id || 'global_default';

    let { data: sessionData } = await supabase
      .from('user_sessions')
      .select('*')
      .eq('session_id', globalSessionId);

    if (!sessionData || sessionData.length === 0) {
      await supabase.from('user_sessions').insert({ session_id: globalSessionId });
      await supabase.from('playlists').insert({
        name: 'default',
        user_id: globalSessionId,
        items: [],
      });

      ({ data: sessionData } = await supabase
        .from('user_sessions')
        .select('*')
        .eq('session_id', globalSessionId));
    }

    return res.status(200).json({
      success: true,
      session: sessionData[0],
    });
  }

  if (req.method === 'OPTIONS') {
    res.setHeader('Access-Control-Allow-Origin', '*');
    res.setHeader('Access-Control-Allow-Methods', 'GET, POST, PUT, DELETE, OPTIONS');
    res.setHeader('Access-Control-Allow-Headers', 'Content-Type');
    return res.status(200).end();
  }

  return res.status(405).json({ error: 'Method not allowed' });
}